File: 백준/Silver/1874. 스택 수열/스택 수열.swift
import Foundation

let n = Int(readLine()!)!

var stack: [Int] = []
var ans: [String] = []
var now: Int = 1

for _ in 1...n {
    let num = Int(readLine()!)!
    
    while now <= num {
        stack.append(now)
        ans.append("+")
        now += 1
    }
    
    if stack.last == num {
        stack.removeLast()
        ans.append("-")
    } else {
        print("NO")
        exit(0)
    }
}

print(ans.joined(separator: "\n"))

File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import java.util.Stack;

public class Main {
     public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        Stack<Integer> stack = new Stack<>();
        List<String> ans = new ArrayList<>();

        int now = 1;
        int n = sc.nextInt();
        boolean flag = false;

        for(int i=0; i<n; i++) {
            int num = sc.nextInt();

            while (now <= num) {
                stack.push(now);
                ans.add("+");
                now++;
            }

            if (stack.lastElement() == num) {
                stack.pop();
                ans.add("-");
            } else {
                System.out.println("NO");
                flag = true;
                break;
            }
        }
        if(!flag) {
            for (int i = 0; i < ans.size(); i++) {
                System.out.println(ans.get(i));
            }
        }
    }
}